### PR TITLE
DEV: Update dependencies to officially support Rails 7.1

### PR DESCRIPTION
**Pull request by Flink**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -35,7 +35,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ruby: ['3.0', '3.1', '3.2']
+        ruby: ['3.1', '3.2', '3.3', '3.4']
```
Comment (CvX) [inline .github/workflows/ci.yml:38]: For consistency with other version lists here 😄 

```suggestion
        ruby: ['3.4', '3.3', '3.2', '3.1']
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,9 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [2.1.0] - 2024-05-29
+- DEV: Update dependencies to officially support Rails 7.1
+
 ## [2.0.1] - 2023-05-30
```
Comment (CvX) [inline CHANGELOG.md:11]: Almost exactly a year later!

File: spec/helpers/url_helper.rb [hunk 3]
```diff
@@ -15,15 +15,9 @@ def get(path)
...
   def flood_get(path, times:)
-    threads = []
-
     times.times do
-      threads << Thread.new do
-        response = get(path)
-        yield response if block_given?
-      end
+      response = get(path)
+      yield response if block_given?
```
Comment (CvX) [inline spec/helpers/url_helper.rb:20]: Does this still actually test "flood get" if it's not doing it, you know, simultaneously? 😅

File: CHANGELOG.md (hunk 2 above)
Comment (Flink) [reply CHANGELOG.md:11]: Indeed!

File: spec/helpers/url_helper.rb (hunk 3 above)
Comment (Flink) [reply spec/helpers/url_helper.rb:20]: Ah, let me check, as there’s a lot of flakiness, I removed the threading part to see if this was improved (it seems a little better), I can check again by reverting the code.

Comment (Flink) [reply spec/helpers/url_helper.rb]: Doesn’t seem to change a lot of things actually :shrug:

**Pull request merged**